pkg/datapath/maps/map.go

```diff
@@ -34,52 +34,73 @@ import (
 
 var (
 	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "datapath-maps")
+
+	globalSweeper = newMapSweeper(&realGarbageWalker{})
 )
 
-// CollectStaleMapGarbage cleans up stale content in the BPF maps from the
-// datapath.
-func CollectStaleMapGarbage() {
-	if option.Config.DryMode {
-		return
-	}
-	walker := func(path string, _ os.FileInfo, _ error) error {
-		return staleMapWalker(path)
-	}
+// garbageWalker checks against its list of the current endpoints to determine
+// whether map paths should be removed, and implements map removal.
+type garbageWalker interface {
+	shouldRemove(id uint16) bool
+	removeMapping(id uint32) error
+	removePath(path string)
```

Comment:
what is `path`?

Reply:
Per the interface comment, a map path. I didn't put a lot of effort into naming things here as it's all confined to this file, but it can be tidied up.

Maybe the context that this abstracts filesystem/endpoint manager access for unit testing would be a useful comment as well, I can add that.